app/http.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Swoole\Atomic;
+use Swoole\Coroutine\Http\Client;
+use Utopia\Logger\Log;
+use Utopia\Logger\Logger;
+use Swoole\Coroutine\Http\Server;
+use Swoole\Http\Request as SwooleRequest;
+use Swoole\Http\Response as SwooleResponse;
+use Swoole\Runtime;
+use Swoole\Table;
+use Swoole\Timer;
+use Utopia\App;
+use Utopia\Balancing\Algorithm\Random;
+use Utopia\Balancing\Algorithm\RoundRobin;
+use Utopia\Balancing\Balancing;
+use Utopia\Balancing\Option;
+use Utopia\CLI\Console;
+use Utopia\Logger\Adapter;
+
+Runtime::enableCoroutine(true, SWOOLE_HOOK_ALL);
+
+$executorStates = new Table(1024);
+$executorStates->column('status', Swoole\Table::TYPE_STRING, 8); // 'online' or 'offline'
+$executorStates->column('state', Swoole\Table::TYPE_STRING, 16384); // State as JSON
+$executorStates->create();
+
+$roundRobinAtomic = new Atomic(0); // Only used if round-robin algo is used
+
+function markOffline(string $executorHostname, string $error, bool $forceShowError = false): void
+{
+    global $executorStates;
+
+    $oldState = $executorStates->exists($executorHostname) ? $executorStates->get($executorHostname) : [];
+
+    $state['status'] = 'offline';
+    $state['hostname'] = $executorHostname;
+    $state['state'] = \json_encode([]);
+    $executorStates->set($executorHostname, $state);
+
+    if (!$oldState || ($oldState['status'] ?? '') === 'online' || $forceShowError) {
+        Console::warning('Executor "' . $executorHostname . '" went down! Message:');
+        Console::warning($error);
+    }
+}
+
+/**
+ * @param array<string, mixed> $state
+ */
+function markOnline(string $executorHostname, array $state, bool $forceShowError = false): void
+{
+    global $executorStates;
+
+    $oldState = $executorStates->exists($executorHostname) ? $executorStates->get($executorHostname) : [];
+
+    $state['status'] = 'online';
+    $state['hostname'] = $executorHostname;
+    $state['state'] = \json_encode($state);
+    $executorStates->set($executorHostname, $state);
+
+    if (!$oldState || ($oldState['status'] ?? '') === 'offline' || $forceShowError) {
+        Console::success('Executor "' . $executorHostname . '" went online.');
+    }
+}
+
+function fetchExecutorsState(bool $forceShowError = false): void
+{
+    $executors = \explode(',', App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        go(function () use ($executor, $forceShowError) {
+            try {
+                $endpoint = 'http://' . $executor . '/v1/health';
+
+                $ch = \curl_init();
+
+                \curl_setopt($ch, CURLOPT_URL, $endpoint);
+                \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+                \curl_setopt($ch, CURLOPT_TIMEOUT, 10);
+                \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+                \curl_setopt($ch, CURLOPT_HTTPHEADER, [
```

Comment:
Worth checking if we can use something a bit more standard here like an `Authorization` header.

Reply:
Used `Authorization Bearer` instead.